Controllers/TeacherController.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Profound.Data;
+using Profound.Data.Models;
+using Profound.Data.ViewModels;
+
+namespace Profound.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class TeacherController : ControllerBase
+    {
+        private readonly IDataRepository _dataRepository;
+
+        public TeacherController(IDataRepository dataRepository)
+        {
+            _dataRepository = dataRepository;
+        }
+
+        [HttpGet("courses")]
+        public IEnumerable<Course> GetCourses()
+        {
+            int teacherId = 1;  // dummy id
+            return _dataRepository.GetCourses().Where(c => c.CreatorId == teacherId);
+        }
+
+        [HttpGet("courses/{courseId}")]
+        public ActionResult<Course> GetCourse(int courseId)
+        {
+            Course course = _dataRepository.GetCourse(courseId);
+
+            if (course == null)
+            {
+                return NotFound();
+            }
+            return course;
+        }
+
+        [HttpPost("courses/create")]
+        public ActionResult<Course> CreateCourse(CourseViewModel courseViewModel)
+        {
+            var course = _dataRepository.CreateCourse(courseViewModel.course);
+            foreach (var category in courseViewModel.CourseCategories)
+            {
+                if (_dataRepository.GetCategory(category.Id) != null)
+                    _dataRepository.InsertToCategoryCourse(course.Id, category.Id);
```

Comment:
As I see it we should stick to one naming convention for create operation.
Create, Post, Insert - we should finally choose the right one and replace all the occurrences.

Or we could just live with the way it is now (:

Reply:
I have changed the name of method "InsertToCategotyCourse". So, the new name is "CreateRecordingForCategoryCourse".

Reply:
Why not just `CreateCategoryCourse`?